core/src/main/scala/io/projectglow/gff/GffFileFormat.scala

```diff
@@ -0,0 +1,328 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.projectglow.gff
+
+import io.projectglow.common.FeatureSchemas._
+import io.projectglow.common.logging.HlsEventRecorder
+import io.projectglow.common.{CompressionUtils, GlowLogging}
+import io.projectglow.gff.GffFileFormat._
+import io.projectglow.sql.util.{HadoopLineIterator, SerializableConfiguration}
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileStatus, Path}
+import org.apache.hadoop.io.compress.{CompressionCodecFactory, SplittableCompressionCodec}
+import org.apache.hadoop.mapreduce.Job
+
+import org.apache.spark.TaskContext
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.execution.datasources.csv.{CSVFileFormat, CSVOptions, UnivocityParser}
+import org.apache.spark.sql.execution.datasources.{DataSource, OutputWriterFactory, PartitionedFile, TextBasedFileFormat}
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.sources.{DataSourceRegister, Filter}
+import org.apache.spark.sql.types.{StringType, StructField, StructType}
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+class GffFileFormat
+    extends TextBasedFileFormat
+    with DataSourceRegister
+    with HlsEventRecorder
+    with GlowLogging {
+
+  private var codecFactory: CompressionCodecFactory = _
+
+  override def shortName(): String = "gff"
+
+  override def isSplitable(
+      sparkSession: SparkSession,
+      options: Map[String, String],
+      path: Path): Boolean = {
+    if (codecFactory == null) {
+      codecFactory = new CompressionCodecFactory(
+        CompressionUtils
+          .hadoopConfWithBGZ(sparkSession.sessionState.newHadoopConf())
+      )
+    }
+
+    // Note: we check if a file is gzipped vs block gzipped during reading, so this will be true
+    // for .gz files even if they aren't actually splittable
+    codecFactory.getCodec(path).isInstanceOf[SplittableCompressionCodec]
+  }
+
+  override def inferSchema(
+      sparkSession: SparkSession,
+      options: Map[String, String],
+      files: Seq[FileStatus]): Option[StructType] = {
+
+    val optionsWithDelimiter = options + ("sep" -> COLUMN_DELIMITER)
+
+    val parsedOptions = new CSVOptions(
+      optionsWithDelimiter,
+      columnPruning = sparkSession.sessionState.conf.csvColumnPruning,
+      sparkSession.sessionState.conf.sessionLocalTimeZone
+    )
+
+    val csv = GffFileFormat
+      .createBaseDataset(sparkSession, files, parsedOptions)
+      .select(attributesField.name)
+      .filter(col(attributesField.name).isNotNull)
+
+    // Use of ParsedAttributesToken helps update sep as soon as detected and avoid looking into
```

Comment:
Auto-detection of tag-value separator with spark sql functions is not straightforward.